extensions/git/src/repository.ts

```diff
@@ -865,13 +865,10 @@ export class Repository implements Disposable {
 	}
 
 	async getInputTemplate(): Promise<string> {
-		const mergeMessage = await this.repository.getMergeMessage();
-		const squashMessage = await this.repository.getSquashMessage();
+		const commitMessage = await this.repository.getMergeMessage() || await this.repository.getSquashMessage();
```

Comment:
No I don't think that's correct. You should use `Promise.all()`. Or even `Promise.race()` since we really only care about whether one of them succeeds.

Reply:
Will look into implementing this now, haven't really used `Promise.race()` but familiarising myself with it. I'll comment here once my commits are final with `Promise.race()` implemented. Sorry about that.

Reply:
Is using `Promise.race()` going to require some refactoring of those two functions and want they return, though? Seems like `Promise.race()` will stop at the first promise resolution or rejection, unless I'm misunderstanding it.

Reply:
Yeah we'd kind of need `[Promise.any](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/any)` which isn't available. Let's maybe just do `Promise.all` and use the first non `undefined` result.

Reply:
How's that (latest commit)? Hopefully this hits the nail on the head!